### PR TITLE
Feat dialog

### PR DESCRIPTION
**Pull request by yeqing123**
Finish feat-dialog.

### CODE REVIEW

File: g [hunk 1]
```diff
@@ -0,0 +1 @@
+3333
```
Comment (yeqing1234) [inline g]: Please modify 3333 to 4444.

**Review by yeqing1234 — COMMENTED** [review]
Please modify 3333 to 4444

File: g (hunk 1 above)
Comment (yeqing123) [reply g]: OK! I have modifyed it.

Comment (yeqing1234) [reply g]: OK ! good!

**Pull request merged**